### ISSUE TITLE
Persisting a @ElementCollection List<String> doesn't seem to work on a Panache Entity

### ISSUE THREAD

**Issue by agoncal** [quarkusio/quarkus#5259]
**Describe the bug**

You can find the entire code here : https://github.com/agoncal/agoncal-sample-quarkus/tree/master/05-PanacheEntities

I have a `Book` Panache entity with a `@ElementCollection List<String>`. After persisting a book with a list of tags, I retrieve the book but the tag list is empty.

The Book Panache entity

```
@Entity
public class Book extends PanacheEntity {

    public String title;
    public String description;
    public Float price;
    public String isbn;
    public Integer nbOfPages;

    @ElementCollection
    public List<String> tags = new ArrayList<>();
   ...
```

The entity is manipulated by a transactional service (not a Panache repository):

```
@ApplicationScoped
@Transactional(SUPPORTS)
public class BookService {

  @Transactional(REQUIRED)
  public Book createBook(Book book) {
    Book.persist(book);
    return book;
  }

  public Book findBook(Long id) {
    return Book.findById(id);
  }
}
```

Then, the test persists a book with 3 tags but once it retrieves it, the tag list is empty 

```
@QuarkusTest
public class BookServiceTest {

  @Inject
  private BookService bookService;

    @Test
    public void shouldCreateABookWithTags() throws Exception {

        Book book = new Book().title("Java EE 7").price(23.5F).isbn("1-84023-742-4").nbOfPages(354);
        book.tag("java ee").tag("java").tag("enterprise");

        // Persists the book to the database
        bookService.createBook(book);

        // Checks the book
        Book foundBook = bookService.findBook(book.id);
        assertEquals(3, foundBook.tags.size());
    }
}
```

**Expected behavior**

This kind of code works fine with a plan Entity without Panache. I've just migrated it to Panache and expect it to work as it is

**Actual behavior**

The tag list is empty.

**Configuration**

I'm using a H2 in memory database

```properties
## Database configuration
quarkus.datasource.url=jdbc:h2:mem:cdbookstoreDB
quarkus.datasource.driver=org.h2.Driver

# drop and create the database at startup (use `update` to only update the schema)
quarkus.hibernate-orm.database.generation=drop-and-create
quarkus.hibernate-orm.log.sql=true
```

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Darwin MacBookeAntonio 18.7.0 Darwin Kernel Version 18.7.0: Tue Aug 20 16:57:14 PDT 2019; root:xnu-4903.271.2~2/RELEASE_X86_64 x86_64

 - Output of `java -version`: java version "1.8.0_212"
Java(TM) SE Runtime Environment (build 1.8.0_212-b10)
Java HotSpot(TM) 64-Bit Server VM (build 25.212-b10, mixed mode)

 - Quarkus version or git rev: 1.0.0.CR1

**Related**

* https://github.com/quarkusio/quarkus/pull/4065
* https://github.com/quarkusio/quarkus/issues/4057

**Comment by FroMage** [comment]
I can only reproduce this in a test class. This code works when invoked in a REST endpoint. So this is a SNEAKY duplicate of #1724. I'll leave it open to make sure we add a test for it when we fix #1724.

**Comment by stuartwdouglas** [comment]
@FroMage is this still an issue? It sounds like it might have been fixed with the new class loading.

**Comment by loicmathieu** [comment]
@agoncal can you test with a recent version of Quarkus? 
This should have been fixed with 1.3.0.

**Comment by agoncal** [comment]
Yes. Just tested with 1.4.1.Final and it works.

Thanks
